### PR TITLE
feat: Add filing Zendesk tickets from the app help

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

When users were submitting support requests via email we lost all the important context that makes debugging much faster.

Additionally we can't do any sort of aggregation in Zendesk, but we sure can in PostHog, e.g. from our Hackathon demo

Screenshot 2023-04-11 at 16 22 36

## Changes

Changed the send an email to be a form that both submits a ticket to Zendesk and also submits an event to PostHog.
We can set up triggers in Zendesk based on "Target area: <something>" and assign them to the appropriate team, see below.

Note: We add a uuid to zendesk and posthog event because we can't otherwise create a link within the Zendesk ticket (don't know the id yet).

The go links can later be changed to be the landing page for viewing the session recordings etc, but currently just resolve to events explorer with some filtering. It's also great because we don't need to change anything about the tickets already filed to update where they link to.

`go/ticket/<zendesk-id>`  <- this can be used in the slack notification or when you know the zendesk ticket it
`go/ticketByUUID/<uuid>` <- this is in the Zendesk ticket
`go/ticket` and `go/ticketByUUID` <- these take us to the events explorer for all ticket events

## How did you test this code?

Filed a ticket from localhost
Screenshot 2023-04-13 at 13 50 09

Screenshot 2023-04-11 at 16 35 48
Screenshot 2023-04-11 at 16 35 56

Which resulted in
A Zendesk ticket: https://posthoghelp.zendesk.com/agent/tickets/2416 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/5384)
and a slack message in #support-pipeline: https://posthog.slack.com/archives/C0460J93NBU/p1681221050450609

Clicked on the go links (e.g. http://go/ticket/2416) and they took me to this [page](http://localhost:8000/events#q=%7B%22kind%22%3A%22DataTableNode%22%2C%22full%22%3Atrue%2C%22source%22%3A%7B%22kind%22%3A%22EventsQuery%22%2C%22select%22%3A%5B%22*%22%2C%22person%22%2C%22properties.zendesk_ticket_link%22%2C%22coalesce(properties.%24current_url%2C%20properties.%24screen_name)%20--%20Url%20%2F%20Screen%22%2C%22properties.%24lib%22%2C%22timestamp%22%5D%2C%22orderBy%22%3A%5B%22timestamp%20DESC%22%5D%2C%22after%22%3A%22-24h%22%2C%22limit%22%3A1%2C%22event%22%3A%22support_ticket%22%2C%22properties%22%3A%5B%7B%22key%22%3A%22zendesk_ticket_id%22%2C%22value%22%3A%5B%222416%22%5D%2C%22operator%22%3A%22exact%22%2C%22type%22%3A%22event%22%7D%5D%7D%2C%22propertiesViaUrl%22%3Atrue%2C%22showSavedQueries%22%3Atrue%7D) (which we can later change to be something more useful), from here our support heroes can see info about the user and view the session recording.

Screenshot 2023-04-11 at 16 16 13
Screenshot 2023-04-11 at 16 02 24

The relevant zendesk triggers: 
1. assigning to pipeline team: https://posthoghelp.zendesk.com/admin/objects-rules/rules/triggers/14405258104347
2. message to support-pipeline:  https://posthoghelp.zendesk.com/admin/objects-rules/rules/triggers/9506745943451

### CODE REVIEW

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: frontend/src/lib/components/HelpButton/HelpButton.tsx
[the remark this answers is not in the document]
Comment (corywatilo) [reply frontend/src/lib/components/HelpButton/HelpButton.tsx:1]: Strongly against pushing people away from asking a question through Squeak! One of the main goals is to _reduce_ the 1:1 support burden. The more public questions/answers we have, the more likely people can self-support.

Also not sure why we'd kill the feedback tool in lieu of this. Why don't we just relabel to _Report a bug_? Leaving under the menu seems fine as we don't want to make it _too_ easy to spin up a ticket when really they may be able to self-support using other methods.

**Comment by tiina303** [comment]
Discussed with @pjhul  and came to the conclusion that we should make it be "Bug / Feedback" (pending @lharries  thoughts on the feedback side). We'll remove the "Ask us a Question" part too as that links to posthog.com/questions, which means we lose all the context (the user likely meant to submit bug/feedback or check the docs).
We don't need nor want bug reports to be a squeak question, it's probably better to just fix the issue. Later once docs are inline we might have access to squeak there, which again will allow us to have all the context needed.

We likely want to remove filing GH issues link too, so we don't lose context and make the menu shorter and nicer.

**Comment by lharries** [comment]
Adding my input as requested by Tiina but realize there's quite a few people giving input here

Agreed with the suggestion of "Bug / Feedback" if it's in the topbar too. Fine to kill the current widget. (We might later want to update the site app to mirror the way we are doing it as other people will likely want to collect bugs separately to feedback too. This could potentially link with @benjackwhite's monitoring plans)

I'd suggest we add in a variant of this code so that the session recording link with the timestamp is added inline to the support ticket and slack ticket. We currently have this with the feedback widget. This helps jump directly to the issue and often comes in handy during longer session recordings

https://github.com/PostHog/posthog/pull/14627/files 

```
const LOOK_BACK = 30
const recordingStartTime = Math.max(
  Math.floor(
      (new Date().getTime() - (posthog?.sessionManager?._sessionStartTimestamp || 0)) / 1000
  ) - LOOK_BACK,
  0
)
const sessionRecordingLink = `${window.location.origin}/recordings/${posthog?.sessionRecording?.sessionId}?t=${recordingStartTime}`
```

For Questions/Squeak: We could still keep the "Question" dropdown within the form but for free users remove the form and instead show a message telling them to add a question on the docs via Squeak?

File: frontend/src/lib/components/Support/supportLogic.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                    message: !message ? 'Please enter a message' : null,
+                    kind: !kind ? 'Please choose' : null,
+                    target_area: !target_area ? 'Please choose' : null,
+                }
+            },
+            submit: async ({ kind, target_area, message }: { kind: string; target_area: string; message: string }) => {
+                actions.submitZendeskTicket(kind, target_area, message)
+                actions.closeSupportForm()
+                // TODO: reset values to empty
+            },
```
Comment (lharries) [inline frontend/src/lib/components/Support/supportLogic.ts:45]: if it's feedback, can we also log it as feedback with this:

```
posthog.capture('Feedback Sent', {
    '$feedback': 'Can you make the logo bigger?'
})
```

File: frontend/src/lib/components/Support/supportLogic.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    const zendesk_ticket_id = res.request.id
+                    const properties = {
+                        zendesk_ticket_uuid,
+                        kind,
+                        target_area,
+                        message,
+                        zendesk_ticket_id,
+                        zendesk_ticket_link: `https://posthoghelp.zendesk.com/agent/tickets/${zendesk_ticket_id}`,
+                    }
+                    posthog.capture('support_ticket', properties)
```
Comment (lharries) [inline frontend/src/lib/components/Support/supportLogic.ts:84]: Will this also be forwarded onto slack?

**Comment by corywatilo** [comment]
> show a message [to free users] telling them to add a question on the docs via Squeak?

Only problem with that is since we de-prioritize responding to free users, we could end up with a higher percentage of unanswered questions which may make the website look like a ghost town.

**Comment by tiina303** [comment]
Reordered the menu (removed GH issue link (no context problem)) and remove question from the form for now (so we can ship this asap and address that in follow-up PRs). Same for adding more links to the form - for recordings and admin panel link to the user - will be done in follow-ups PRs.

**Review by timgl — APPROVED** [review]

**Review by benjackwhite — CHANGES_REQUESTED** [review]
I think this needs to replace the "Feedback" button in the top bar before it can be merged personally. Once we have that removed, then I'm fine with this

**Review by benjackwhite — APPROVED** [review]

**Pull request merged**